### PR TITLE
Add support for python 3.13

### PR DESCRIPTION
**Pull request by king-phyte**
### Other changes:
- Drop support for python 3.8
- Update dependencies

### CODE REVIEW

**Comment by king-phyte** [comment]
@aminalaee Please take a look at this PR (and maybe squash and merge 😅)

**Comment by trim21** [comment]
look good to me, no actually code changes

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -68,20 +68,20 @@ jobs:
       - uses: actions/checkout@v3
```
Comment (trim21) [inline .github/workflows/ci.yml:68]: should also upgrade this one

**Comment by trim21** [comment]
`actions/upload-artifact@v4` is not broken for matrix, it just different name.

for example, use `wheels-${{ matrix.job-index }}` and use `wheels-*` in download-artifact v4

**Comment by king-phyte** [comment]
> for example, use `wheels-${{ matrix.job-index }}` and use `wheels-*` with `merge-multiple` (or another option name, I forget) in download-artifact v4

@trim21 Correct me if I am wrong, but I think you mean `strategy.job-index` [from the docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#strategy-context)

**Comment by king-phyte** [comment]
@trim21 can we get this reviewed and merged?

**Comment by trim21** [comment]
> @trim21 can we get this reviewed and merged?

I don't have permission to merge PR

**Comment by king-phyte** [comment]
Oh. My bad. Thanks for the review and nudges by the way @trim21 
I suppose we have to wait for @aminalaee

**Comment by trim21** [comment]
is there a classifiers for Python39+ ？

**Comment by king-phyte** [comment]
Not that I know of

**Review by aminalaee — APPROVED** [review]
LGTM. Thanks both for the PR

**Comment by king-phyte** [comment]
@aminalaee Thank you. If I may ask, when can we expect the new release. This is currently blocking one of my projects from transitioning to Python 3.13

**Comment by trim21** [comment]
I think we should at least solve https://github.com/aminalaee/uuid-utils/issues/69 by merging https://github.com/aminalaee/uuid-utils/pull/70 or remove `SafeUUID` from pyi

**Pull request merged**